### ISSUE TITLE
(Possibly) erroneous PipError: Double requirement given

### ISSUE THREAD

**Issue by malikoth** [di/pip-api#39]
OK, here's my setup:

`base.txt`
```
django
```

`test.txt`
```
-r base.txt
pytest
```

`dev.txt`
```
-r base.txt
-r test.txt
werkzeug
```

When attempting to parse `dev.txt`, `pip_api` errors with 
```
pip_api.exceptions.PipError: Double requirement given: django (already in django, name='django')
```

However, I feel that this is a bug, as running `pip install -r dev.txt` works just fine, and what's more, it only shows the following line once in its output:
```
Requirement already satisfied: django in /usr/local/lib/python3.7/site-packages (from -r base.txt (line 1)) (2.2.2)
```

That makes me believe that pip is parsing all of these requirements files, and successfully removing duplicate requirements.  So I believe that this `PipError` is in ... well, uh, error.  It seems awfully deliberate though, so I'm not sure what problem that exception is trying to solve, when it could just silently skip over duplicate requirements, like pip seems to.

Again, happy to submit a PR, but I'm not sure what the correct approach would be to fixing this.  Thanks!

**Comment by di** [comment]
I can't reproduce this. Here's what I did:

```
$ tree
.
├── base.txt
├── dev.txt
└── test.txt

0 directories, 3 files

$ cat base.txt
django
blah $ cat dev.txt
-r base.txt
-r test.txt
werkzeug

$ cat test.txt
-r base.txt
pytest

$ python
Python 3.7.0 (default, Jul 30 2018, 11:52:05)
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip_api
>>> pip_api.parse_requirements('dev.txt')
{'werkzeug': <Requirement('werkzeug')>, 'django': <Requirement('django')>, 'pytest': <Requirement('pytest')>}
>>>
```

Seems to work correctly to me? What versions of python/pip/pip-api are you using?

**Comment by malikoth** [comment]
OK, it's my fault.  First of all, there was a bug in my previous PR.  Second of all, I simplified the test case too far when writing the test for it before.  Third, I simplified the test case when trying to explain this bug too far as well.  Fourth, I didn't dig into this bug far enough before posting.

You will see the error if you put all three files described above into a `reqs` folder, and then attempt to parse `reqs/dev.txt`.

Incoming PR with fix.
